File: python/hackerrank/math/kaprekarNumbers.py

def kaprekarNumbers(p, q):
    """
    Finds and prints all modified Kaprekar numbers in the range [p, q] inclusive.
    
    A modified Kaprekar number is a positive integer n such that when it is squared, 
    the sum of the two parts of the square (where the right part has the same number 
    of digits as n) equals n.
    
    Args:
        p (int): The lower integer limit.
        q (int): The upper integer limit.
        
    Returns:
        None: Prints the Kaprekar numbers as a space-separated string, or 
              "INVALID RANGE" if none are found.
    """
    results = []
    for n in range(p, q + 1):
        d = len(str(n))
        square, divisor = n * n, 10**d
        
        right, left = square % divisor, square // divisor
        
        if left + right == n:
            results.append(n)
            
    if results:
        print(*(results))
    else:
        print("INVALID RANGE")

def main():
    try:
        p = int(input().strip())
        q = int(input().strip())
        kaprekarNumbers(p, q)
    except EOFError:
        pass

if __name__ == '__main__':
    main()

File: golang/hackerrank/math/kaprekarNumbers.go
package math
import (
	"fmt"
	"math"
	"strconv"
)








// KaprekarNumbers finds and prints all modified Kaprekar numbers in the range [p, q].
func KaprekarNumbers(p, q int) {
	var results []int
	for n := p; n <= q; n++ {
		d := len(strconv.Itoa(n))
		square := int64(n) * int64(n)
		divisor := int64(math.Pow(10, float64(d)))

		right := square % divisor
		left := square / divisor

		if left+right == int64(n) {
			results = append(results, n)
		}
	}

	if len(results) > 0 {
		for i, v := range results {
			fmt.Print(v)
			if i < len(results)-1 {
				fmt.Print(" ")
			}
		}
		fmt.Println()
	} else {
		fmt.Println("INVALID RANGE")
	}
}
